### PR TITLE
microbit: Enable sign-in as query variant

### PR DESCRIPTION
**Pull request by eanders-ms**
For microbit, enable sign-in with url parameter `identity=1`. This will allow us to test in beta, etc.

### CODE REVIEW

**Review by jwunderl — APPROVED** [review]

File: pxtarget.json [hunk 1]
```diff
@@ -646,6 +624,35 @@
             "appTheme": {
                 "disableBlobObjectDownload": true
             }
+        },
+        "identity=1": {
+            "cloud": {
```
Comment (kimprice) [inline pxtarget.json:629]: Why do you need the extra cloud field?

Comment (kimprice) [reply pxtarget.json:629]: oh nvm I see it's the same above. Just needed to expand.

File: pxtarget.json [hunk 2]
```diff
@@ -31,29 +31,7 @@
...
-            },
-            "google": {
-                "id": "google",
-                "name": "Google",
-                "icon": "/static/providers/google-logo.svg",
-                "identity": true,
-                "redirect": true,
-                "order": 2
-            }
+            "github": {}
```
Comment (kimprice) [inline pxtarget.json:34]: why do you need this line?

File: pxtarget.json (hunk 1 above)
Comment (jwunderl) [reply pxtarget.json:629]: the way queryVariant works is that it applies a patch to the appTarget (the rest of this json blob) -- if you go to arcade and open the devtools you can see `pxt.appTarget.cloud.cloudProviders` contains similar / the same data as this to enable sign in, what this does is for microbit.makecode.com/beta?identity=1 it'll override the default value (the other part changed in this pr) for `pxt.appTarget.cloud.cloudProviders` with the one inside this object

File: pxtarget.json (hunk 2 above)
Comment (eanders-ms) [reply pxtarget.json:34]: Reverted my earlier changes, moving them to the "queryVariants" section where they'll be merged into the root of this object when `identity=1` is provided.

File: pxtarget.json (hunk 1 above)
Comment (eanders-ms) [reply pxtarget.json:629]: It's to match the schema of this config object. The JSON snippet specified in this override will be merged into the root of this object when the `identity=1` parameter is present.

**Pull request merged**